clamr fails to build with gcc-8.0.1

```
[ 29%] Building CXX object crux/CMakeFiles/pcrux.dir/crux.cpp.o
cd /tmp/ci/spack-stage/spack-stage-PXVdjp/CLAMR/spack-build/crux && /home/ci/spack/lib/spack/env/gcc/g++  -DHAVE_CONFIG_H -DHAVE_MPI -DMIXED_PRECISION -I/home/ci/spack/var/spack/stage/clamr-master-tqfbejmjaw4527dau4gzfq5zr6ea3hvj/CLAMR -I/usr/include/openmpi-x86_64 -I/usr/lib64/openmpi/include -I/tmp/ci/spack-stage/spack-stage-PXVdjp/CLAMR/spack-build  -ftree-vectorize -ftree-vectorize -O2 -g -DNDEBUG -fPIC   -o CMakeFiles/pcrux.dir/crux.cpp.o -c /home/ci/spack/var/spack/stage/clamr-master-tqfbejmjaw4527dau4gzfq5zr6ea3hvj/CLAMR/crux/crux.cpp
In file included from /usr/include/c++/8/map:60,
                 from /home/ci/spack/var/spack/stage/clamr-master-tqfbejmjaw4527dau4gzfq5zr6ea3hvj/CLAMR/MallocPlus/MallocPlus.h:56,
                 from /home/ci/spack/var/spack/stage/clamr-master-tqfbejmjaw4527dau4gzfq5zr6ea3hvj/CLAMR/MallocPlus/MallocPlus.cpp:59:
/usr/include/c++/8/bits/stl_tree.h: In instantiation of 'class std::_Rb_tree<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, malloc_plus_memory_entry*>, std::_Select1st<std::pair<const std::__cxx11::basic_string<char>, malloc_plus_memory_entry*> >, cmp_str, std::allocator<std::pair<const std::__cxx11::basic_string<char>, malloc_plus_memory_entry*> > >':
/usr/include/c++/8/bits/stl_map.h:151:17:   required from 'class std::map<std::__cxx11::basic_string<char>, malloc_plus_memory_entry*, cmp_str>'
/home/ci/spack/var/spack/stage/clamr-master-tqfbejmjaw4527dau4gzfq5zr6ea3hvj/CLAMR/MallocPlus/MallocPlus.cpp:95:48:   required from here
/usr/include/c++/8/bits/stl_tree.h:452:21: error: static assertion failed: comparison object must be invocable with two arguments of key type
       static_assert(__is_invocable<_Compare&, const _Key&, const _Key&>{},
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [MallocPlus/CMakeFiles/pMallocPlus.dir/build.make:66: MallocPlus/CMakeFiles/pMallocPlus.dir/MallocPlus.cpp.o] Error 1
make[2]: Leaving directory '/tmp/ci/spack-stage/spack-stage-PXVdjp/CLAMR/spack-build'
make[1]: *** [CMakeFiles/Makefile2:2558: MallocPlus/CMakeFiles/pMallocPlus.dir/all] Error 2
```
Details here: https://travis-ci.org/proxyapps/proxy-ci/jobs/375671286 and https://api.travis-ci.org/v3/job/375671286/log.txt

--- a/MallocPlus/MallocPlus.h
+++ b/MallocPlus/MallocPlus.h
@@ -100,9 +100,13 @@ struct malloc_plus_memory_entry {
 
 struct cmp_str
 {
-   bool operator()(char const *a, char const *b)
+   bool operator()(char const *a, char const *b) const
    {
       return strcmp(a, b) < 0;
+   }
+   bool operator()(const string &a, const string &b) const
+   {
+      return strcmp(&a[0], &b[0]) < 0;
    }
 };
 

